hw/ip/otbn/dv/rig/rig/gens/bad_branch.py

```diff
@@ -26,42 +26,7 @@ class BadBranch(SnippetGen):
     ends_program = True
 
     def __init__(self, cfg: Config, insns_file: InsnsFile) -> None:
-        super().__init__()
-
-        self.insns = []
-        self.weights = []
-
-        self.beq = self._get_named_insn(insns_file, 'beq')
-        self.bne = self._get_named_insn(insns_file, 'bne')
-
-        # beq and bne expect operands: grs1, grs2, offset
-        for insn in [self.beq, self.bne]:
-            if not (len(insn.operands) == 3 and
-                    isinstance(insn.operands[0].op_type, RegOperandType) and
-                    insn.operands[0].op_type.reg_type == 'gpr' and
-                    not insn.operands[0].op_type.is_dest() and
-                    isinstance(insn.operands[1].op_type, RegOperandType) and
-                    insn.operands[1].op_type.reg_type == 'gpr' and
-                    not insn.operands[1].op_type.is_dest() and
-                    isinstance(insn.operands[2].op_type, ImmOperandType) and
-                    insn.operands[2].op_type.signed):
-                raise RuntimeError('{} instruction from instructions file is not '
-                                   'the shape expected by the Branch generator.'
-                                   .format(insn.mnemonic))
-
-        self.imm_op_type = self.bne.operands[2].op_type
-
-        for insn in insns_file.insns:
-            if insn.mnemonic in ['beq', 'bne']:
-                weight = cfg.insn_weights.get(insn.mnemonic)
-                if weight > 0:
-                    self.weights.append(weight)
-                    self.insns.append(insn)
-
-        # Check that at least one instruction has a positive weight.
-        assert len(self.insns) == len(self.weights)
-        if not self.weights:
-            self.disabled = True
+        super().__init__(cfg, insns_file)
```

Comment:
This is fine, but you can actually get rid of the `__init__` function entirely if it's just calling the superclass constructor.